### PR TITLE
prevent catwalks connecting to grilles

### LINKED ISSUE #10733: grilles auto-connect to catwalks
### Labels

[minor]

### Description

As title. Presumably because catwalks are weird grilles internally.

### Steps to reproduce

1. Go on donut3
2. See all the under-window grilles connecting to catwalks nearby

### Expected Behavior

No autoconnect

### Additional Information & Screenshots

image
Easiest to see with the south-facing connection

### PR DESCRIPTION
**Pull request by TheTyrant25**
[GAME OBJECT]
## About the PR 
Prevents grilles and catwalks auto connecting to each other. This is done by primarily moving the code for catwalks and grilles apart under a new abstract parent: `/obj/mesh`. `grille` and `catwalk` are now both subtypes of `mesh`.

This PR also replaces every use instance of `/obj/grille/catwalk` and `/obj/grille` to be correctly pathed.

## Why's this needed?
Fixes #10733

### CODE REVIEW

**Comment by TheTyrant25** [comment]
!merge upstream

**Comment by Tarmunora** [comment]
Hm. Kinda feel like this would be best handled by refactoring out an abstract parent, so that catwalks aren't a child of grilles and `connect_to` can be used as expected without having to is istype for carveouts

**Comment by TheTyrant25** [comment]
fair enough. I'll do that instead then

**Comment by TheTyrant25** [comment]
massive find replace moment
also, i do not know how to use updatepaths, so any advice in setting that up for this pr would be appreciated

File: code/obj/grille.dm [hunk 1]
```diff
@@ -504,7 +379,7 @@
 				icon_state = "grille[builtdir]" + "-0"
 
 	proc/update_neighbors()
-		for (var/obj/grille/G in orange(1,src))
+		for (var/obj/mesh/grille/G in orange(1,src))
 			G.UpdateIcon()
```
Comment (Tarmunora) [inline code/obj/grille.dm:383]:
```suggestion
		for (var/obj/mesh/O in orange(1,src))
			O.UpdateIcon()
```

File: code/obj/grille.dm [hunk 2]
```diff
@@ -589,3 +464,137 @@
...
+	icon_state = "C15-0"
+	density = FALSE
+	desc = "This doesn't look very safe at all!"
+	layer = CATWALK_LAYER
+	shock_when_entered = FALSE
+	auto = TRUE
+	plane = PLANE_FLOOR
+	amount_of_rods_when_destroyed = 1
+	var/catwalk_type = "C" // Short for "Catwalk"
+	var/connects_to = list(/obj/mesh/catwalk, /obj/machinery/door) // We're working differently from grilles. We don't check a list and then another, we check all possible atoms to connect to.
```
Comment (Tarmunora) [inline code/obj/grille.dm:502]: Unused var

**Review by Tarmunora — CHANGES_REQUESTED** [review]
`obj/mesh/update_icon()` implementation should either be written in a general way that works for both grilles and catwalks, or the current implementation should be pulled down to `obj/mesh/grille/update_icon()`

Catwalks do not work, probably due to marked problems

May be more issue yet to reveal themselves

also maps in secrets will need path updates

**Comment by Tarmunora** [comment]
Catwalks still do not work

**Comment by TheTyrant25** [comment]
damn

**Comment by TheTyrant25** [comment]
this has gotten really bad really fast. I can't be bothered to fix this merge conflict. I'll revisit it later

**Pull request closed without merging**